blueocean-dashboard/src/main/js/PipelineRoutes.jsx

```diff
@@ -82,9 +85,16 @@ function persistModalBackground() {
  * the background context.
  */
 function handleNavigationChangeToFromModal(prevState, nextState, replace, callback) {
-    if (nextState.params.runId && (prevState == null || !prevState.params.runId)) {
+    const enterRunDetails = nextState.params.runId && (prevState == null || !prevState.params.runId);
+    const leaveRunDetails = (prevState !== null && prevState.params.runId) && !nextState.params.runId;
+    const enterCreatePipeline = nextState.location.pathname.indexOf('/create-pipeline') !== -1 &&
+        (prevState == null || prevState.location.pathname.indexOf('/create-pipeline') === -1);
+    const leaveCreatePipeline = nextState.location.pathname.indexOf('/create-pipeline') === -1 &&
+        (prevState == null || prevState.location.pathname.indexOf('/create-pipeline') !== -1);
```

Comment:
@cliffmeyers hey Cliff, what are we trying to determine in the above code? It's a sea of `location.pathname.indexOf('/create-pipeline')` .... maybe we can compose this a bit more .... boil it down a bit and put it into a function(s) with some comments/javadoc etc